modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,260 @@
+----
+
+Notice that the `age` field is the only one that is optional, as indicated by the `?`.
+
+You can use the type to know if a value follows the structure defined by the JSON schema.
+The following example outputs the value `true` because the object contains the required fields, `firstName` and `lastName`:
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ lastName: "Doe"
+} is Root
+----
+
+.Output:
+----
+"true"
+----
+
+The following example outputs the value `false` since the object does not contain the required fields, `firstName` and `lastName`:
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ age: 20
+} is Root
+----
+
+.Output:
+----
+"false"
+----
+
+== Use Definitions Inside Schemas
+
+Definitions inside JSON schemas are also loaded when the module is imported:
+
+.JSON Schema (`example/schema/Account.json`):
+[source,JSON]
+----
+{
+ "$schema": "http://json-schema.org/draft-04/schema#",
+ "definitions": {
+   "address": {
+     "type": "object",
+     "properties": {
+       "street_address": {
+         "type": "string"
+       },
+       "city": {
+         "type": "string"
+       },
+       "state": {
+         "type": "string"
+       }
+     },
+     "required": [
+       "street_address",
+       "city",
+       "state"
+     ]
+   },
+   "person": {
+     "type": "object",
+     "properties": {
+       "firstName": {
+         "type": "string",
+         "description": "The person's first name."
+       },
+       "lastName": {
+         "type": "string",
+         "description": "The person's last name."
+       },
+       "age": {
+         "description": "Age in years which must be equal to or greater than zero.",
+         "type": "integer",
+         "minimum": 0
+       }
+     },
+     "required": [
+       "firstName",
+       "lastName"
+     ]
+   }
+ },
+ "type": "object",
+ "properties": {
+   "person": {
+     "$ref": "#/definitions/person"
+   },
+   "address": {
+     "$ref": "#/definitions/address"
+   }
+ }
+}
+----
+
+You can import the types from that schema with the following directive, assuming the JSON schema is `example/schema/Account.json`:
+
+----
+import * from jsonschema!example::schema::Account
+----
+
+The types from that schema have the same effect as declring the following types:
+
+----
+type Root = { address?: address, person?: person }
+
+type address = { street_address: String, city: String, state: String }
+
+type person = { firstName: String, lastName: String, age?: Number }
+----
+
+Use the import directive to import a single type from the schema:
+
+----
+import address from jsonschema!example::schema::Account
+----
+
+To avoid type name collision, you can use the `as` keyword to change the imported type name to use another name:
```

Comment:
```suggestion
To avoid type-name collision, you can use the `as` keyword to change the imported type name to use another name:
```

Reply:
(Here, you correctly use the article with the keyword name as an adjective.)

Reply:
Reworded.